src/components/SpeciesDeliverableTable/AddSpeciesModal.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { Grid, useTheme } from '@mui/material';
+import { SelectT } from '@terraware/web-components';
+
+import DialogBox from 'src/components/common/DialogBox/DialogBox';
+import TextField from 'src/components/common/Textfield/Textfield';
+import Button from 'src/components/common/button/Button';
+import { useOrganization } from 'src/providers';
+import { useParticipantData } from 'src/providers/Participant/ParticipantContext';
+import { SpeciesService } from 'src/services';
+import {
+  ParticipantProjectSpecies,
+  ParticipantProjectSpeciesRequest,
+} from 'src/services/ParticipantProjectSpeciesService';
+import strings from 'src/strings';
+import { Species } from 'src/types/Species';
+
+export interface AddEditSubLocationProps {
+  onClose: () => void;
+  participantProjectSpecies: ParticipantProjectSpecies[];
+  reload: () => void;
+}
+
+export default function AddSpeciesModal(props: AddEditSubLocationProps): JSX.Element {
+  const { onClose, participantProjectSpecies, reload } = props;
+  const { currentParticipantProject } = useParticipantData();
+  const { selectedOrganization } = useOrganization();
+  const [allSpecies, setAllSpecies] = useState<Species[]>();
+  const [selectableSpecies, setSelectableSpecies] = useState<Species[]>();
+  const [participantProjectSpeciesIds, setParticipantProjectSpeciesIds] = useState<number[]>();
+  const [selectedSpecies, setSelectedSpecies] = useState<Species>();
+  const [error, setError] = useState<string>('');
+  const theme = useTheme();
+
+  useMemo(() => {
+    const ids = participantProjectSpecies.map((ppSpecies) => ppSpecies.id);
+    setParticipantProjectSpeciesIds(ids);
+  }, [participantProjectSpecies]);
+
+  useEffect(() => {
+    const populateSpecies = async () => {
+      const response = await SpeciesService.getAllSpecies(selectedOrganization.id);
+      if (response.requestSucceeded) {
+        setAllSpecies(response.species);
+      }
+    };
+
+    void populateSpecies();
+  }, [selectedOrganization.id, location]);
+
+  const initializeSpeciesProject = () => {
+    if (currentParticipantProject) {
+      return {
+        projectId: currentParticipantProject.id,
+        speciesId: -1,
+      };
+    }
+  };
+
+  useEffect(() => {
+    const speciesToAdd = allSpecies?.filter((species) => {
+      return !participantProjectSpeciesIds?.includes(species.id);
+    });
+    setSelectableSpecies(speciesToAdd);
+  }, [allSpecies]);
+
+  const [record] = useState<ParticipantProjectSpeciesRequest | undefined>(initializeSpeciesProject());
+
+  const save = () => {
+    if (selectedSpecies) {
+      setError('');
+      // const newProjectSpecies = { ...record, speciesId: selectedSpecies.id };
+      // save newProjectSpecies
```

Comment:
@joshbuchea @nickgraz I see we have the selectParticipantProjectSpeciesUpdateRequest created, but don't we need a selectParticipantProjectSpeciesCreateRequest?

Reply:
Hmm that is correct. Not sure how I missed that

Reply:
Are you able to add it?

Reply:
Sure, I will add it.